blockchain/vm/internaltx_tracer.go

```diff
@@ -515,64 +514,13 @@ func (this *InternalTxTracer) result() (*InternalTxTrace, error) {
 	}
 	if err := this.ctx["error"]; err != nil && err.(error).Error() == errEvmExecutionReverted.Error() {
 		outputHex := this.ctx["output"].(string) // it is already a hex string
-		outputHexLength := len(outputHex)
-
-		// outputHexLength should be equal for larger than 138 (10+32*2+32*2) to parse a revert string
-		// outputHex[:10]: "0x08c379a0" == crypto.Keccak256([]byte("Error(string)"))[:4]
-		// outputHex[10:10+32*2]: a revert string
-		// outputHex[10+32*2:10+32*2+32*2]: the length of a revert string
-		if outputHexLength >= 138 && outputHex[2:10] == "08c379a0" {
-			defaultOffset := 10
-
-			// TODO-Klaytn: introduce error handling logic for the case the parsing data is bigger than math.MaxUint64
-			stringOffset, err := strconv.ParseUint(outputHex[defaultOffset+32*2-8:defaultOffset+32*2], 16, 64)
-			if err != nil {
-				logger.Error("failed to parse hex string to get stringOffset",
-					"err", err, "outputHex", outputHex)
-				return nil, err
-			}
-			stringLength, err := strconv.ParseUint(outputHex[defaultOffset+32*2+32*2-8:defaultOffset+32*2+32*2], 16, 64)
-			if err != nil {
-				logger.Error("failed to parse hex string to get stringLength",
-					"err", err, "outputHex", outputHex)
-				return nil, err
-			}
-
-			if stringOffset > uint64(outputHexLength) {
-				stringOffset = 0
-			}
-			if stringLength > uint64(outputHexLength) {
-				stringLength = 0
-			}
-
-			start := defaultOffset + 32*2 + int(stringOffset*2)
-			end := start + int(stringLength*2)
-
-			if start < 0 {
-				start = 0
-			}
-			if end < 0 {
-				end = 0
-			}
-			// return nothing if end is out of the range
-			if end > outputHexLength {
-				start = outputHexLength - 1
-				end = outputHexLength - 1
-			}
 
-			// left-padding with 0 for hex decoding
-			outputHexSlice := outputHex[start:end]
-			if len(outputHexSlice)%2 == 1 {
-				outputHexSlice = "0" + outputHexSlice
-			}
-			asciiInBytes, err := hex.DecodeString(outputHexSlice)
-			if err != nil {
-				logger.Error("failed to parse hex string to get ASCII representation",
-					"err", err, "outputHex", outputHex)
-				return nil, err
```

Comment:
It seems that there is no error case in this method. How about removing `error` to be returned?

Reply:
I am okay with not returning errors from the private function `result()`. I will add a commit.

@jeongkyun-oh But should we stop returning errors from the public function `GetResult()`?
I'd say we keep returning errors for the following reasons:
- If we add another Go tracer, the tracer might return errors.
- For consistency with JS tracer https://github.com/klaytn/klaytn/blob/v1.9.0/node/cn/tracers/tracer.go#L631
- For compatibility with Eth https://github.com/ethereum/go-ethereum/blob/v1.10.21/eth/tracers/tracers.go#L36

Reply:
@blukat29 It makes sense. In my side, a block should be processed without returning an error because this causes bad block. I think updating only private function is enough in this PR. I will update the outer function if you agree.

Reply:
@jeongkyun-oh To avoid bad block we can simply modify `ApplyTransaction()`? https://github.com/klaytn/klaytn/blob/v1.9.0/blockchain/blockchain.go#L2628.

```diff
	var internalTrace *vm.InternalTxTrace
	if vmConfig.EnableInternalTxTracing {
		internalTrace, err = GetInternalTxTrace(vmConfig.Tracer)
		if err != nil {
			logger.Error("failed to get tracing result from a transaction", "txHash", tx.Hash().String(), "err", err)
-			return nil, 0, nil, err
		}
	}
```

I think the caller should deal with it. Block processors will ignore the error and debug APIs will return the error.

Reply:
@blukat29 Yes, you are right. This is simplest one. How about panic? Do you think that we need to recover the panic?

Reply:
@jeongkyun-oh Methods like `CaptureState()` and `GetResult()` is good places to recover.
But after the previous and present fixes, I don't see any possible panic in InternalTxTracer.
While recovering from the panic is no harm, I am also fine without recovery for the sake of simplicity.

Reply:
@blukat29 Thanks for sharing your idea.

Reply:
Then,` this.err` did not have effect until now? Which means every callers `GetResult()` does not need to take and consider the error (this [PR](https://github.com/klaytn/klaytn/pull/1584) as well)

Reply:
@hyunsooda `this.err` is often populated when (1) tracing interrupted or (2) `this.step()` fails, (3) execution faults. 
Among those, case (1) can happen and is relevant to debug APIs ([code](https://github.com/klaytn/klaytn/blob/v1.9.0/node/cn/api_tracer.go#L782)), case (2) never happens in the current code, case (3) can happen when executing a malformed contract (e.g. ErrInvalidJump).
Therefore `this.err` is necessary even after this PR.

Reply:
Uh oh, I was confused that the `this.err` and returned error of the private function `result()` are the same thing. Oops.. I'm so sorry for the noise. Thank you.